pkg/app/piped/controller/planner.go

```diff
@@ -325,3 +345,38 @@ func (p *planner) reportDeploymentCancelled(ctx context.Context, commander, reas
 	}
 	return err
 }
+
+func (p *planner) getMentionedAccounts(ctx context.Context, event model.NotificationEventType, targetDSP deploysource.Provider) ([]string, error) {
+	var spec config.GenericDeploymentSpec
+
+	if targetDSP == nil {
+		repo := p.deployment.GitPath.Repo
+		gitRepo, err := p.gitClient.Clone(ctx, repo.Id, repo.Remote, repo.Branch, "")
+		if err != nil {
+			return nil, fmt.Errorf("failed to clone repository %s: %w", repo.Id, err)
+		}
+
+		absPath := filepath.Join(gitRepo.GetPath(), p.deployment.GitPath.GetDeploymentConfigFilePath())
+		cfg, err := config.LoadFromYAML(absPath)
+		if err != nil {
+			if os.IsNotExist(err) {
+				return nil, fmt.Errorf("deployment config file %s was not found", absPath)
+			}
+			return nil, err
+		}
+		spec = cfg.KubernetesDeploymentSpec.GenericDeploymentSpec
+	} else {
```

Comment:
Thank you. I know why we need this.
I just feel things are becoming complicated time by time. So let me find a better way to retrieve Deployment Configuration instead of that way.

Currently, every deployment are handled by componnents in order as: `trigger` -> `planner` -> `executor`
And we need the Deployment Configuration at someplace in the `executor`, now we need it in the `planner` right, next we may need it in the `trigger` too.

So how about load the Deployment Configuration just in `trigger` and save somewhere (e.g. in deployment model 🤔 ) to use by all places.
https://github.com/pipe-cd/pipe/tree/master/pkg/app/piped/trigger

Reply:
I like that! Can I create another PR and replace all these notification parts with it? The way is easy for me.

Reply:
@ono-max Sure. We can merge this PR and do it in another PR.
So how about removing the cloning part for this time? We only deal with the case when `targetDSP` is not nil.
https://github.com/pipe-cd/pipe/pull/2552/files#diff-cae5b5638e4b9ea22e482124c2a91059d78b4417ba4bdd958717f20e77aac20fR352-R368